pkg/personname/groupInfo_test.go

```diff
@@ -0,0 +1,444 @@
+				NamePrefix: "MR",
+				NameSuffix: "",
+			},
+		},
+		// No prefix or trailing
+		{
+			Raw: "CROUCH^BARTEMIUS^'BARTY'",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "'BARTY'",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+		},
+		// No middle or trailing
+		{
+			Raw: "CROUCH^BARTEMIUS",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "BARTEMIUS",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+		},
+		// No given or trailing
+		{
+			Raw: "CROUCH",
+			Expected: GroupInfo{
+				FamilyName: "CROUCH",
+				GivenName:  "",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+		},
+		// Middle Only with missing prefix separator
+		{
+			Raw: "^^'BARTY'^",
+			Expected: GroupInfo{
+				FamilyName:        "",
+				GivenName:         "",
+				MiddleName:        "'BARTY'",
+				NamePrefix:        "",
+				NameSuffix:        "",
+				TrailingNullLevel: GroupNullLevelPrefix,
+			},
+		},
+		// No family or trailing
+		{
+			Raw: "",
+			Expected: GroupInfo{
+				FamilyName: "",
+				GivenName:  "",
+				MiddleName: "",
+				NamePrefix: "",
+				NameSuffix: "",
+			},
+			IsEmpty: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		// Test creating a new GroupInfo object
+		t.Run(tc.Raw+"_New", func(t *testing.T) {
+			newGroup := GroupInfo{
+				tc.Expected.FamilyName,
+				tc.Expected.GivenName,
+				tc.Expected.MiddleName,
+				tc.Expected.NamePrefix,
+				tc.Expected.NameSuffix,
+				tc.Expected.TrailingNullLevel,
+			}
+
+			dcm, err := newGroup.DCM()
+			if err != nil {
+				t.Fatal("DCM() returned error:", err)
+			}
+
+			if tc.Raw != dcm {
+				t.Errorf(
+					"formatted .DCM() does not match input: "+
+						"expected '%v', got '%v'",
+					tc.Raw,
+					dcm,
+				)
+			}
+		})
+
+		// Test parsing a group string.
+		t.Run(tc.Raw+"_Parse", func(t *testing.T) {
+
+			parsed, err := groupFromValueString(tc.Raw, pnGroupAlphabetic)
+			if err != nil {
+				t.Fatal("error parsing value:", err)
+			}
+
+			fmt.Println("EXPECTED NULLS:", tc.Expected.TrailingNullLevel)
+			checkGroupInfo(t, tc.Expected, parsed, tc.Raw, "")
+
+			dcm, err := parsed.DCM()
+			if err != nil {
+				t.Fatal("DCM() returned error:", err)
+			}
+
+			if tc.Raw != dcm {
+				t.Errorf(
+					"formatted .DCM() does not match input: "+
+						"expected '%v', got '%v'",
+					tc.Raw,
+					dcm,
+				)
+			}
+		})
+
+		// Test .IsEmpty() method.
+		t.Run(tc.Raw+"_IsEmpty", func(t *testing.T) {
+			parsed, err := groupFromValueString(tc.Raw, pnGroupAlphabetic)
+			if err != nil {
+				t.Fatalf("error parsing value '%v': %v", tc.Raw, err)
+			}
+
+			if tc.IsEmpty != parsed.IsEmpty() {
+				t.Errorf(
+					".IsEmpty() returned %v, extected %v",
+					parsed.IsEmpty(),
+					tc.IsEmpty,
+				)
+			}
+		})
+	}
+}
+
+// TestGroupInfo_DCM_interiorNullsExceedTrailingLevel tests that if a group has interior
+// null values that exceed the
```

Comment:
nit: unfinished comment

Reply:
Ha, thanks for catching those!

I decided to go with "[INVALID]" (updated both this time) so the brackets make it a little more clear it's a meta-value, not a value-value.